### ISSUE TITLE
Add random_state (as keyword argument?) to generate_folds and use it when executing stratified sampling

### ISSUE THREAD

**Issue by MarcelloGiannini** [RasaHQ/rasa#12373]
https://github.com/RasaHQ/rasa/blob/8e786957508dd0c23ddf81440d02409ff2107ca1/rasa/nlu/test.py#L1450-L1456

In order to allow replicability of results while cross validating, it would be useful to be able to provide the random_state variable as input of generate_folds and then use it when calling sklearn.model_selection.StratifiedKFold. Right now any cross validation would randomly shuffle the dataset and provide different folds every time generate_folds is called.

**Comment by m-vdb** [comment]
Hi there, thanks for raising this. I kindly ask you to create your issue on Jira: https://rasa-open-source.atlassian.net/jira/software/c/projects/OSS/boards/1 . Thanks!
